### PR TITLE
Many superficial fixes to the contributor book

### PR DESCRIPTION
**Pull request by louisfd**
I did an extensive review of the contributor book. I think once this is merged it will be ready to be hosted

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ license = "MIT OR Apache-2.0"
 [workspace.dependencies]
 async-trait = "0.1.79"
 bytemuck = "1.14"
-candle-core = { version = "0.4.1" }
+candle-core = { git = "https://github.com/huggingface/candle.git", branch = "main" }
```
Comment (antimora) [inline Cargo.toml:28]: Any reason for this change?

File: crates/burn-candle/src/backend.rs [hunk 2]
```diff
@@ -104,6 +105,7 @@ impl<F: FloatCandleElement, I: IntCandleElement> Backend for Candle<F, I> {
             candle_core::Device::Metal(device) => {
                 // For some reason, device.wait_until_completed() does not seem to work,
                 // and neither does writing and reading a value with into_data
+                // See this issue: https://github.com/huggingface/candle/issues/2050
```
Comment (antimora) [inline crates/burn-candle/src/backend.rs:108]: Probably Candle related changes are not meant to be here.

**Review by antimora — CHANGES_REQUESTED** [review]
The text looks good but there are candle related changes that were probably not meant here.

File: Cargo.toml (hunk 1 above)
Comment (louisfd) [reply Cargo.toml]: No, my bad

**Comment by louisfd** [comment]
@antimora Thanks for the review. Fixed the unrelated candle stuff and also the run-checks command which was wrong.

**Review by antimora — APPROVED** [review]
LGTM

**Pull request merged**